### DISCUSSION TITLE
Using --unsquash with singularity

### DISCUSSION THREAD

**Discussion by kenneditodd** [nextflow-io/nextflow#4117] [Q&A]
Hello,

I am trying to run the [MOP2 Nextflow pipeline](https://github.com/biocorecrg/MOP2). I am new to using nextflow but am learning a lot as I go. I am submitting jobs to a cluster using slurm. The admins intentionally left squashfuse off the node. I keep getting the error message below.
```
FATAL:   container creation failed: mount hook function failure: mount /proc/self/fd/3->/usr/local/biotools/apptainer/apptainer-1.1.4/var/apptainer/mnt/session/rootfs error: while mounting image /proc/self/fd/3: squashfuse not found
```
I see that there is a `--unsquash` option with singularity exec. I am trying to add the unsquash argument to the apptainer scope with runOptions. However, I'm not sure the best place to put this. There are two potential config files to add this to - I have a [nextflow.global.config](https://github.com/biocorecrg/MOP2/blob/main/nextflow.global.config) and a [slurm.config](https://github.com/biocorecrg/MOP2/blob/main/conf/slurm.config). 

Where would I add the following code so I can submit to SLURM? I think it may look like one of the two ideas shown below but have had no success. Could be an issue with my syntax or placement.
```
containerOptions =  'unsquash'
```
or
```
apptainer {
     runOptions = '--unsquash'
}
```

**Comment by bentsherman** [comment]
You probably need to use `apptainer.engineOptions`. Either config file is fine, just depends on whether you want it to apply to all runs with apptainer or only with your SLURM system.
